aea/cli/registry/utils.py

```diff
@@ -36,7 +38,78 @@ def request_api(method, path, params=None):
         return resp.json()
     elif resp.status_code == 403:
         raise click.ClickException('You are not authenticated.')
+    elif resp.status_code == 404:
+        raise click.ClickException('Not found in Registry.')
     else:
         raise click.ClickException(
             'Wrong server response. Status code: {}'.format(resp.status_code)
         )
+
+
+def split_public_id(public_id):
+    """Split public ID to ownwer, name, version."""
+    public_id = public_id.replace(':', '/')
+    return public_id.split('/')
+
+
+def _download_file(url, cwd):
+    """Download file from URL and save it in CWD (current working directory)."""
+    local_filename = url.split('/')[-1]
+    filepath = os.path.join(cwd, local_filename)
+    # NOTE the stream=True parameter below
+    response = requests.get(url, stream=True)
+    if response.status_code == 200:
+        with open(filepath, 'wb') as f:
+            f.write(response.raw.read())
+    else:
+        raise click.ClickException(
+            'Wrong response from server when downloading package.'
+        )
+    return filepath
+
+
+def _extract(source, target):
+    """Extract tarball and remove source file."""
+    if (source.endswith("tar.gz")):
+        tar = tarfile.open(source, "r:gz")
+        tar.extractall(path=target)
+        tar.close()
+    else:
+        raise Exception('Unknown file type: {}'.format(source))
+
+    os.remove(source)
+
+    folder_name = os.path.basename(source).replace('.tar.gz', '')
+    extracted_folder = os.path.join(target, folder_name)
+    return extracted_folder
+
+
+def fetch_package(obj_type, public_id, cwd):
```

Comment:
Alle these methods are missing docstrings listing the params and the params have no type annotations.